### PR TITLE
Encode HTML entities inside placeholder when using double braces

### LINKED ISSUE #4: Encoding as HTML Entities to avoid code injection
If the output is for a browser, then it can be a security issue (code injection, etc.) if the values are simply dumped straight into the output - rather, they should be encoded as HTML entities. To that end, it's annoying doing it oneself before passing data to the templating engine. Is it planned to add support for this directly to pupa?

e.g. Maybe single braces `{}` means do it normally, and double braces (`{{}}`) could mean "escape it too please"? Or even the other way around.

### PR DESCRIPTION
**Pull request by apsknight**
This PR fix #4.

Encode HTML entities inside placeholder to avoid code injection when using double braces.

- [x] Update `index.js` to escape HTML.
- [x] Update `test.js` to include tests for the double braces. 
- [x] Update `README` with double braces feature.

The code of HTML escaping is inspired from [this answer](https://stackoverflow.com/a/30970751) on Stackoverflow.

### CODE REVIEW

**Comment by apsknight** [comment]
@sindresorhus Please review this PR.

File: index.js [hunk 1]
```diff
@@ -9,6 +9,30 @@ module.exports = (template, data) => {
...
+
+	if (HTMLregex.test(template)) {
+		template = template.replace(HTMLregex, (_, key) => {
+			let ret = data;
+			for (const prop of key.split('.')) {
+				ret = ret ? ret[prop] : '';
+			}
+
+			// Encoding HTML Entities to avoid code injection
+			const HTMLentities = /[&"<>]/g;
```
Comment (sindresorhus) [inline index.js]: You can use https://github.com/sindresorhus/escape-goat for escaping

**Comment by sindresorhus** [comment]
There's a lot of duplicate logic. Would be nice to reuse some of it.

File: readme.md [hunk 2]
```diff
@@ -27,6 +27,13 @@ pupa('The mobile number of {name} is {phone.mobile}', {
 
 pupa('I like {0} and {1}', ['🦄', '🐮']);
 //=> 'I like 🦄 and 🐮'
+
+// Double braces encodes the HTML entities to avoid code injection.
+pupa('{{0}}{{1}}', ['<br>yo</br>', '<i>lol</i>']);
+//=> '&lt;b&gt;yo&lt;/b&gt;&lt;b&gt;lol&lt;/b&gt;
+
+pupa('yo {{foo}} lol {{bar}} sup', {foo: '🦄', bar: '🌈'});
+//=> 'yo 🦄 lol 🌈 sup's
```
Comment (sindresorhus) [inline readme.md]: One example is enough. This can be removed.

File: readme.md [hunk 3]
```diff
@@ -27,6 +27,13 @@ pupa('The mobile number of {name} is {phone.mobile}', {
 
 pupa('I like {0} and {1}', ['🦄', '🐮']);
 //=> 'I like 🦄 and 🐮'
+
+// Double braces encodes the HTML entities to avoid code injection.
+pupa('{{0}}{{1}}', ['<br>yo</br>', '<i>lol</i>']);
+//=> '&lt;b&gt;yo&lt;/b&gt;&lt;b&gt;lol&lt;/b&gt;
```
Comment (sindresorhus) [inline readme.md]: I would prefer reusing the example on line 28, just wrap the emojis in HTML tags.

**Comment by apsknight** [comment]
@sindresorhus 	I've added the changes as suggested by you. Please review the updated PR.

Thanks!

**Pull request merged**